server/contexts/ctxerr/statistics.go

```diff
@@ -0,0 +1,49 @@
+package ctxerr
+
+import (
+	"context"
+	"encoding/json"
+)
+
+type errorAgg struct {
+	Count int      `json:"count"`
+	Loc   []string `json:"loc"`
+}
+
+// Aggregate retrieves all errors in the store and returns an aggregated,
+// json-formated summary containing:
+// - The number of ocurrences of each error
+// - A reduced stack trace used for debugging the error
+func Aggregate(ctx context.Context) (json.RawMessage, error) {
+	empty := json.RawMessage("[]")
+	maxTraceLen := 3
+
+	storedErrs, err := Retrieve(ctx)
+	if err != nil {
+		return empty, Wrap(ctx, err, "retrieve on aggregation")
+	}
+
+	aggs := make([]errorAgg, len(storedErrs))
+	for i, stored := range storedErrs {
+		var em FleetErrorChainJSON
+		if err = json.Unmarshal(stored.Error, &em); err != nil {
+			return empty, Wrap(ctx, err, "unmarshal on aggregation")
+		}
+
+		// build a full stack trace
+		stack := em.Cause.Stack
+		for _, wrap := range em.Wraps {
+			stack = append(stack, wrap.Stack...)
```

Comment:
Somewhat a nit, but you should check for the maxTraceLen here and ensure you stop appending (and exit the loop) when you reach the limit, so that you don't unnecessarily append and then throw away extra stack lines.

Maybe something along the lines of (untested):

```
remaining := maxTraceLen - len(stack)
if remaining > 0 {
    max := len(wrap.Stack)
    if max > remaining {
        max = remaining
    }
    stack = append(stack, wrap.Stack[:max]...)
} else {
    break
}
```

Reply:
gotcha, this was me being lazy because I thought "how long can a stack trace be"? but agreed exiting earlier is the right way to go. Pushed a fix, lmk what do you think

Reply:
Yeah for sure it will be quite small, not a huge deal. Thanks for updating!